x-pack/plugins/security_solution/common/api/detection_engine/model/rule_schema/rule_request_schema.test.ts

```diff
@@ -5,59 +5,48 @@
  * 2.0.
  */
 
-import * as t from 'io-ts';
-import { left } from 'fp-ts/lib/Either';
-import { pipe } from 'fp-ts/lib/pipeable';
-import { exactCheck, foldLeftRight, getPaths } from '@kbn/securitysolution-io-ts-utils';
-
+import { expectParseError, expectParseSuccess, stringifyZodError } from '@kbn/zod-helpers';
 import { getListArrayMock } from '../../../../detection_engine/schemas/types/lists.mock';
-import type { SavedQueryRuleCreateProps } from './rule_schemas';
-import { RuleCreateProps } from './rule_schemas';
 import {
+  getCreateEsqlRulesSchemaMock,
```

Comment:
Looks like new schema is more permissive and allows supplying invalid fields in request body.

Old schema

```bash
curl --location --request POST 'http://localhost:5601/kbn/api/detection_engine/rules' \
--header 'kbn-xsrf: kibana' \
--header 'elastic-api-version: 2023-10-31' \
--header 'Authorization: Basic INSERT_YOUR_CREDS' \
--header 'Content-Type: application/json' \
--data-raw '{
    "type": "esql",
    "language": "esql",
    "query": "from packetbeat-8.5.3 | stats counted = count_distinct(destination.domain)",
    "false_positives": [],
    "references": [],
    "risk_score": 100,
    "risk_score_mapping": [],
    "severity": "low",
    "severity_mapping": [],
    "name": "ee",
    "description": "sample rule",
    "tags": [],
    "license": "",
    "interval": "45m",
    "from": "now-2760s",
    "to": "now",
    "meta": {
        "from": "1m",
        "kibana_siem_app_url": "http://localhost:5601/kbn/app/security"
    },
    "actions": [],
    "random_field": 4
}'
{"statusCode":400,"error":"Bad Request","message":"[request body]: invalid keys \"random_field\""}%

```


New schema

```bash
curl --location --request POST 'http://localhost:5601/kbn/api/detection_engine/rules' \
--header 'kbn-xsrf: kibana' \
--header 'elastic-api-version: 2023-10-31' \
--header 'Authorization: Basic INSERT_YOUR_CREDS' \
--header 'Content-Type: application/json' \
--data-raw '{
    "type": "esql",
    "language": "esql",
    "query": "from packetbeat-8.5.3 | stats counted = count_distinct(destination.domain)",
    "false_positives": [],
    "references": [],
    "risk_score": 100,
    "risk_score_mapping": [],
    "severity": "low",
    "severity_mapping": [],
    "name": "ee",
    "description": "sample rule",
    "tags": [],
    "license": "",
    "interval": "45m",
    "from": "now-2760s",
    "to": "now",
    "meta": {
        "from": "1m",
        "kibana_siem_app_url": "http://localhost:5601/kbn/app/security"
    },
    "actions": [],
    "random_field": 4
}'
{"name":"ee","description":"sample rule","risk_score":100,"severity":"low","license":"","output_index":"","meta":{"from":"1m","kibana_siem_app_url":"http://localhost:5601/kbn/app/security"},"version":1,"tags":[],"enabled":true,"risk_score_mapping":[],"severity_mapping":[],"interval":"45m","from":"now-2760s","to":"now","actions":[],"exceptions_list":[],"author":[],"false_positives":[],"references":[],"max_signals":100,"threat":[],"id":"f296c8a0-728c-11ee-a021-21ca0041a607","rule_id":"ee28ae5b-7ed4-44d0-ac14-6b6f99776b69","immutable":false,"updated_at":"2023-10-24T16:46:59.563Z","updated_by":"elastic","created_at":"2023-10-24T16:46:59.563Z","created_by":"elastic","revision":0,"related_integrations":[],"required_fields":[],"setup":"","type":"esql","language":"esql","query":"from packetbeat-8.5.3 | stats counted = count_distinct(destination.domain)"}

```

Reply:
Yes, that was intentional. We now allow extra fields in schemas for forward compatibility, but we remove them from the payload during parsing. So from now on, extra fields are simply ignored and won't lead to validation errors.

Reply:
> Yes, that was intentional. We now allow extra fields in schemas for forward compatibility, but we remove them from the payload during parsing. 

What forward compatibility it allows? Accepting any random field in request doesn't look as designed to make it compatible with next versions, but rather loose validation. 
If we consider ES API example: it does not allow invalid random keys, but it does not mean  in future it won't support new key in request, when new functionality will be added

Reply:
> What forward compatibility it allows?

@vitaliidm  Forward compatibility in API design usually means the ability to accept payload from future version of a API client. For the particular case of detection rules, it means that we should be able introduce changes to the rule schema without breaking interoperability with older versions of stack.

The TRADE team has been requesting forward compatible APIs for a long time already, that will allow them to use newer rules with older Kibana without issues.